vaadin-testbench-shared/src/main/java/com/vaadin/testbench/ElementQuery.java

```diff
@@ -301,10 +484,34 @@ public ElementQuery<T> withoutAttribute(String attribute) {
      * @return this element query instance for chaining
      *
      * @see #withoutAttributeContaining(String, String)
+     * @see #withoutAttributeContainingToken(String, String)
      */
     public ElementQuery<T> withoutAttribute(String attribute, String value) {
-        attributes.add(new AttributeMatch(attribute, "!=", value));
-        return this;
+        return withAttribute(attribute, value, NOT_MATCHES_EXACTLY);
+    }
+
+    /**
+     * Selects on elements not having the given attribute containing the given token.
```

Comment:
`token` could be misleading here
```suggestion
     * Selects on elements not having the given attribute containing the given text.
```

Reply:
Good catch!